### PR TITLE
Fixes #24261 - pagination CSS is wrong on assign hosts for taxonomy

### PR DESCRIPTION
**Pull request by boaz0**
The reason why this happens is because there is a nested form:
the host pagination nested under hosts assign.

Because of that the form tag in pagination is not rendered.
To fix this, the form tag in pagination was replaced by div.

Why?:
It seems like the pagination does not have to be a form. It
can be a div element that its select handles the onchange event
which calls `turbolinks.visit`.

/cc @ohadlevy @lizagilman

### CODE REVIEW

**Comment by ripcurld** [comment]
[test katello]

**Comment by ripcurld** [comment]
[test foreman]

**Comment by boaz0** [comment]
[test foreman]

**Comment by lizagilman** [comment]
LGTM :+1:

File: test/integration/organization_js_test.rb [hunk 1]
```diff
@@ -15,4 +15,16 @@ class OrganizationJSTest < IntegrationTestWithJavascript
       end
     end
   end
+
+  test "test per page in hosts assign" do
+    (1..10).map{ Host.create }
```
Comment (tbrisker) [inline test/integration/organization_js_test.rb]: you can use `FactoryBot.create_list(:host, 10, organization: nil)` instead

**Review by tbrisker — CHANGES_REQUESTED** [review]
This breaks the ability to type a page number in the pagination. Also small comment inline.

**Comment by ripcurld** [comment]
[test foreman]

**Comment by ripcurld** [comment]
[test foreman]

**Comment by ripcurld** [comment]
[test foreman]

**Comment by ohadlevy** [comment]
@boaz1337 whats the status of this PR? thanks.

**Comment by boaz0** [comment]
@tbrisker - can you review this change, please? :pray:

**Comment by boaz0** [comment]
https://github.com/theforeman/foreman/commit/d4e81e9610de1d1802d2747d47c4badd60e5150d fixes this.

**Pull request closed without merging**